### PR TITLE
Migrate to View binding, SDK 34 and Version catalogs

### PR DESCRIPTION
**Pull request by naveensingh**
### Major changes:
 - Migrate build scripts from Groovy to [Kotlin DSL](https://developer.android.com/build/migrate-to-kotlin-dsl)
  - Migrate to [Gradle version catalogs](https://developer.android.com/build/migrate-to-catalogs)
 - Migrate from Kotlin Kapt to [KSP (Kotlin symbol processing)](https://developer.android.com/build/migrate-to-ksp) as Kapt is now in maintenance mode.
 - Migrate from Kotlin synthetics to [View binding](https://developer.android.com/topic/libraries/view-binding).
 - Target SDK 34:
   - [SCHEDULE_EXACT_ALARM](https://developer.android.com/reference/android/Manifest.permission#SCHEDULE_EXACT_ALARM) was removed on API 33 and above in favor of [USE_EXACT_ALARM](https://developer.android.com/reference/android/Manifest.permission#USE_EXACT_ALARM) because it's a normal permission and granted on install but apps are not allowed to use this permission unless their core functionality depends on it (this app qualifies). Whenever`AlarmManager.canScheduleExactAlarm()` returns false,`setAndAllowWhileIdle()` is used to ensure the functionality isn't broken. This should only happen on Android 12 when the user revokes this permission manually.
   - No other changes are required.

### Minor changes:
 - Fix paddings in the landscape year fragment layout, it's aligned now and the months start in the same row.
 - Remove `item_autocomplete_email_binding.xml`, `item_autocomplete_email_name.xml` in favor of `item_autocomplete_title_subtitle.xml`

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt [hunk 1]
```diff
@@ -1375,13 +1383,13 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
...
         val dayCode = Formatter.getDayCodeFromTS(timestamp / 1000L)
-        calendar_fab.beVisible()
+        binding.calendarFab.beVisible()
         config.storedView = DAILY_VIEW
         updateViewPager(dayCode)
     }
 
     // events fetched from Thunderbird, https://www.thunderbird.net/en-US/calendar/holidays and
-    // https://holidays.kayaposoft.com/public_holidays.php?year=2021
+    // https://holidays.kayaposoft.com/publicHolidays.php?year=2021
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt:1392]: this shouldnt have been edited

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt [hunk 2]
```diff
@@ -43,7 +43,7 @@ import kotlin.math.min
 import kotlin.math.roundToInt
 
 class WeekFragment : Fragment(), WeeklyCalendar {
-    private val WEEKLY_EVENT_ID_LABEL = "event_id_label"
+    private val WEEKLY_EVENT_ID_LABEL = "eventIdLabel"
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt:46]: guess this wasnt supposed to be edited

Comment (naveensingh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt:46]: yeah, I may have accidentally renamed a few things. I'll rename them before marking this for review...

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt (hunk 1 above)
Comment (naveensingh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt]: fixed it

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt (hunk 2 above)
Comment (naveensingh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt]: fixed that too

**Comment by tibbi** [comment]
the displayed day count at the bottom right corner of weekly view doesnt change colors on light/dark theme

**Comment by naveensingh** [comment]
I'm checking it...

**Comment by naveensingh** [comment]
Fixed it

**Comment by tibbi** [comment]
ok nice thanks, havent found other issues so will accept it

**Pull request merged**